doc/src/manual/embedding.md

```diff
@@ -284,6 +285,62 @@ args[1] = some_other_value;
 JL_GC_POP();
 ```
 
+Each scope must have only one call to `JL_GC_PUSH*`. Hence, if all variables cannot be pushed once by
+a single call to `JL_GC_PUSH*`, or if there is more than 6 variables to be pushed and using an array
+of arguments is not an option, then one can use inner blocks:
+
+```c
+jl_value_t *ret1 = jl_eval_string("sqrt(2.0)");
+JL_GC_PUSH1(&ret1);
+jl_value_t *ret2 = 0;
+{
+    jl_function_t *func = jl_get_function(jl_base_module, "exp");
+    ret2 = jl_call1(func, ret1);
+    JL_GC_PUSH1(&ret2);
+    // Do something with ret2.
+    JL_GC_POP();    // This pops ret2.
+}
+JL_GC_POP();    // This pops ret1.
+```
+
+If it is required to hold the pointer to a variable between functions (or block scopes), then it is
+not possible to use `JL_GC_PUSH*`. In this case, it is necessary to create and keep a reference to the
+variable in the Julia global scope. One simple way to accomplish this is to use a global `IdDict` that
+will hold the references, avoiding deallocation by the GC. However, this method will only work
+properly with mutable types. If the variable is immutable, then it needs to be wrapped in an
+equivalent mutable container (like `Vector`) before it is pushed to `IdDict`. In this approach, the C
+code must store the pointer to the container, and not to the variable itself. For very simple
```

Comment:
Errrrr, C code can store the pointer to the variable itself just fine. You just need to make sure any pointers you passed to julia code for this purpose must be the wrapper. I.e. the wrapper must be constructed or filled in C code and unless you dig into the `IdDict` internal, you must store the wrapper in it.

Reply:
> then we will have the same problem as described before right?

Nop. The only issue is pointer identity, i.e. the pointer recieved by the julia function is not guaranteed to be the same **pointer** even though it is guaranteed to be the same **object**. For keeping the pointer live this is obviously a problem but for using the object this is totally fine. For normal use, you pretty much only need to tell julia which object you are using, in most case the actual julia code might not even see your pointer or use any pointers at all...

Note that the `Vector{Any}` or `RefValue{Any}` has to be created/filled in via C code and not by using `jl_call*`. You can use `jl_new_struct`. Or more generally, the one stored in the container and the one stored in the C code has to be synced via C code directly. You can certainly create the container object via julia code and use C code to (re)load the pointer to be used in C code but that's more complicated than necessary.

Reply:
`RefValue`*, Seems correct otherwise.

Reply:
Sorry 😅 I will update the PR. Thanks very much for all the clarifications @yuyichao !